afpacket/afpacket.go

```diff
@@ -248,16 +250,17 @@ func (h *TPacket) releaseCurrentPacket() error {
 func (h *TPacket) ZeroCopyReadPacketData() (data []byte, ci gopacket.CaptureInfo, err error) {
 	h.mu.Lock()
 retry:
-	if h.current == nil || !h.current.next() {
+	if h.current == nil || !h.headerNextNeeded || !h.current.next() {
 		if h.shouldReleasePacket {
 			h.releaseCurrentPacket()
 		}
 		h.current = h.getTPacketHeader()
```

Comment:
@gconnell thx, you're right code around getTPacketHeader needs to be reworked too. I'll try to have a look too.